### PR TITLE
refac: add TableRows class for holding data from databricks

### PR DESCRIPTION
**Pull request by JesperGomes**
## References

* #1023

### CODE REVIEW

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/CalculationResultClient.cs [hunk 1]
```diff
@@ -165,13 +165,15 @@ private string ToDeltaValue(TimeSeriesType timeSeriesType)
...
-                    QuantityQualityMapper.MapQuality(res[2])))
-            .ToArray();
-
-        return new ProcessStepResult(timeSeriesType, pointsDto);
+        var tableRows = databricksSqlResponse.TableRows;
+
+        var pointsDto = Enumerable.Range(0, databricksSqlResponse.TableRows.Count)
+            .Select(row => new TimeSeriesPoint(
+                DateTimeOffset.Parse(tableRows[row, ResultColumnNames.Time]),
+                decimal.Parse(tableRows[row, ResultColumnNames.Quantity], CultureInfo.InvariantCulture),
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/CalculationResultClient.cs:173]: It's a bit involved to get the desired value from the row. Are you planning on encapsulating the complexity for reusability?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/DatabricksSqlResponse.cs [hunk 2]
```diff
@@ -14,4 +14,4 @@
 
 namespace Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.CalculationResultClient;
 
-public sealed record DatabricksSqlResponse(string State, IEnumerable<string[]> Rows);
+public sealed record DatabricksSqlResponse(string State, TableRows TableRows);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/DatabricksSqlResponse.cs:17]: I prefer the shorter name
```suggestion
public sealed record DatabricksSqlResponse(string State, TableRows Rows);
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/ResultColumnNames.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.CalculationResultClient;
+
+public static class ResultColumnNames
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/ResultColumnNames.cs:17]: We should add a contract test to ensure refactoring in pyspark doesn't break these naming assumptions.
In another PR?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/DatabricksSqlResponseParserTests.cs [hunk 4]
```diff
@@ -35,7 +35,7 @@ public DatabricksSqlResponseParserTests()
 
     [Theory]
     [AutoMoqData]
-    public void Create_ReturnsResponseWithCorrectState(DatabricksSqlResponseParser sut)
+    public void Parse_ReturnsResponseWithCorrectState(DatabricksSqlResponseParser sut)
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/DatabricksSqlResponseParserTests.cs:38]: General suggestion: How about the following to stick to the already used term "expected" inside all test implementations?
```suggestion
    public void Parse_ReturnsResponseWithExpectedState(DatabricksSqlResponseParser sut)
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/DatabricksSqlResponseParserTests.cs [hunk 5]
```diff
@@ -79,30 +79,33 @@ public void Create_ReturnsDataArrayWithCorrectContent(DatabricksSqlResponseParse
         var actual = sut.Parse(_sampleJson);
 
         // Assert
-        actual.Rows.First().Should().Equal(expectedFirstArray);
+        actual.TableRows[0].Should().Equal(expectedFirstArray);
+        actual.TableRows[^1].Should().Equal(expectedLastArray);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/DatabricksSqlResponseParserTests.cs:83]: Nice. I didn't know `[^1]`.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/DatabricksSqlResponseParserTests.cs [hunk 6]
```diff
@@ -79,30 +79,33 @@ public void Create_ReturnsDataArrayWithCorrectContent(DatabricksSqlResponseParse
...
-        var obj = new JObject(
-            new JProperty("status", new JObject(new JProperty("state", "PENDING"))),
-            new JProperty("result", new JObject(new JProperty("data_array", new List<string[]>()))));
+        var status = new JProperty("status", new JObject(new JProperty("state", "PENDING")));
+        var manifest = new JProperty("manifest", new JObject(new JProperty("schema", new JObject(new JProperty("columns", new JArray(new JObject(new JProperty("name", "grid_area"))))))));
+        var result = new JProperty("result", new JObject(new JProperty("data_array", new List<string[]>())));
+        var obj = new JObject(status, manifest, result);
 
         // Act + Assert
         sut.Parse(obj.ToString()!);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/DatabricksSqlResponseParserTests.cs:97]:
```suggestion
        sut.Parse(obj.ToString()!).Should().NotBeNull();
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/DatabricksSqlResponseParserTests.cs [hunk 7]
```diff
@@ -79,30 +79,33 @@ public void Create_ReturnsDataArrayWithCorrectContent(DatabricksSqlResponseParse
...
-        var obj = new JObject(
-            new JProperty("not_status", new JObject(new JProperty("state", "PENDING"))),
-            new JProperty("result", new JObject(new JProperty("data_array", new List<string[]>()))));
+        var status = new JProperty("not_status", new JObject(new JProperty("state", "PENDING")));
+        var manifest = new JProperty("manifest", new JObject(new JProperty("schema", new JObject(new JProperty("columns", new JArray(new JObject(new JProperty("name", "grid_area"))))))));
+        var result = new JProperty("result", new JObject(new JProperty("data_array", new List<string[]>())));
+        var obj = new JObject(status, manifest, result);
 
         // Act + Assert
         Assert.Throws<InvalidOperationException>(() => sut.Parse(obj.ToString()));
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/DatabricksSqlResponseParserTests.cs:111]: And you are (always) 100% sure that it's not the `obj.ToString()` that throws the exception? 😉

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/ResultColumnNames.cs (hunk 3 above)
Comment (JesperGomes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/ResultColumnNames.cs:17]: yes, I will add a task in Zenhub

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/TableRowsTests.cs [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    [Fact]
+    public void IndexerWithInvalidRowIndex_ThrowsException()
+    {
+        // Arrange
+        var sut = new TableRows(_columnNames, _rowData);
+
+        // Act + Assert
+        Assert.ThrowsAny<Exception>(() => sut[100, "col2"]);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Infrastructure/CalculationResultClient/TableRowsTests.cs:91]: General idea: How about not making me check these values by being explicit? And then have the reusable valid/invalid values defined close to the other class level data.
```suggestion
        // Arrange
        var sut = new TableRows(_columnNames, _rowData);

        // Act + Assert
        Assert.ThrowsAny<Exception>(() => sut[_invalidRowIndex, _validColumnName]);
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/TableRows.cs [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// namespace Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.CalculationResultClient;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.CalculationResultClient;
+
+public class TableRows
```
[the remark this answers is not in the document]
Comment (JesperGomes) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/CalculationResultClient/TableRows.cs:18]: Yes, I kept changing the name. Maybe you are right.

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**